### ISSUE TITLE
`dumps_task` in `SimpleShuffleLayer` and `BroadcastJoinLayer` unpack

### ISSUE THREAD

**Issue by gjoseph92** [dask/dask#7650]
Just a TODO that `SimpleShuffleLayer` and `BroadcastJoinLayer` both use `dumps_task` in their `__dask_distributed_unpack__` methods, which leads to pickling on the scheduler, which we want to avoid.

Additionally, could this manual use of `pickle.dumps` in `BroadcastJoinLayer.__dask_distributed_pack__` lead to a double-pickle scenario, where when the task actually gets executed, the kwarg values are bytestrings, not the actual unpickled values: https://github.com/dask/dask/blob/b3a86469cb339767617c03c05e60db101ceb177a/dask/layers.py#L663-L671

Related to https://github.com/dask/distributed/pull/4699.

cc @rjzamora @madsbk

**Comment by rjzamora** [comment]
I totally agree that we should try to move away from `dumps_task` in cases where it requires the task `callable`  to handle serialization at run time. Just some notes...

>which leads to pickling on the scheduler, which we want to avoid.

Right - we would like to avoid any use of pickle on the scheduler (in case pickle is disabled entirely), however the main concern is that we cannot call `pickle.loads` on the scheduler.  The cases you have highlighted above do not require `pickle.loads`.

>Additionally, could this manual use of pickle.dumps in BroadcastJoinLayer.__dask_distributed_pack__ lead to a double-pickle scenario, where when the task actually gets executed, the kwarg values are bytestrings, not the actual unpickled values

Yes - This is exactly what is expected to happen now.  This is why the functions used within the broadcast-join tasks currently check if anything needs to be unpickled on the worker.  We avoided this issue in `Blockwise` by relying on `to_serialize`.
